### DISCUSSION TITLE
How to transfer a vector field or its components

### DISCUSSION THREAD

**Discussion by QuangDom** [idaholab/moose#20110] [Q&A General] [answered]
Dear MOOSE Team,

I am working with INSAD module, and would like to transfer the velocity (velocity in INSAD is a vector
 field) into a sub-app. When I am using MultiAppProjectionTransfer, I got this error:

```
sub0: *** ERROR ***
sub0: The following error occurred in the object "MOOSE Problem", of type "FEProblem".
sub0: 
sub0: No standard variable named vel found. Did you specify a vector variable when you meant to specify a standard variable?
```

Is there any transfer support this purpose? Or is there anyway to transfer the components of velocity (velocity_x, velocity_y)  separately?

Thanks in advance,
/ Quang

**Comment by GiudGiud** [comment, marked as answer]
Hello

So vector/array variable transfers is not uniformly supported. It's a project of ours. 
For now,  you'll need to transfer the components separately,
1) use a vector component aux to store each component in a separate auxvariable 
https://mooseframework.inl.gov/source/auxkernels/VectorVariableComponentAux.html

2) use the transfer you want on each component aux-variable

3) two scenarios:
- if you need this in a regular variable form (so each component is its own variable), then you can use what you just transferred
- if you need this back in vector form, then we need to create a vector equivalent of this: https://mooseframework.inl.gov/source/auxkernels/BuildArrayVariableAux.html. 
We currently dont have this for for your use case. 

It should not be too hard to make though.

Guillaume

**Comment by PengWei97** [comment]
Hi @GiudGiud,

I had the same problem, whether the vector auxVariable at each nodel in main-app can be transferred via some `transfer` into the sub-app. Or, so far, there is still no suitable transfer object to solve this problem.

Any help would be greatly appreciated!

Thanks,

Wei

**Reply by GiudGiud** [reply]
Hello

You ll need to decompose the vector variable into its components and reform it after


We have very few transfers that support vector variables

Guillaume

**Reply by PengWei97** [reply]
Ok, thanks a lot.
